src/payment/strategies/paypal-commerce/paypal-commerce-sdk.ts

```diff
@@ -60,12 +61,162 @@ export interface ButtonsOptions {
     style?: PaypalButtonStyleOptions;
     fundingSource?: string;
     createOrder?(): Promise<string>;
-    onApprove?(data: ApproveDataOptions): void;
+    onApprove?(data: ApproveDataOptions, actions: ApproveActions): void;
+    onShippingChange?(data: ShippingChangeData, actions: ApproveActions): void;
     onClick?(data: ClickDataOptions, actions: ClickActions): void;
     onCancel?(): void;
     onError?(error: Error): void;
 }
 
+export interface ApproveActions {
+    order: {
+        capture(): Promise<PayerDetails>;
+        authorize(): Promise<PayerDetails>;
+        patch(data: PatchArgument[]): void;
+    };
+    resolve(): void;
+    reject(): void;
+}
+
+export interface CheckoutWithBillingAddress {
+    consignments: Consignment[];
+    cart: {
+        id: string;
+        lineItems: {
+            physicalItems: PhysicalItem[];
+        };
+    };
+}
+
+interface PhysicalItem {
+    id: string;
+    imageUrl: string;
+}
+
+interface PatchArgument {
+    op: string;
+    path: string;
+    value: PatchValue | ShippingOptions[];
+}
+
+interface PatchValue {
+    currency_code: string;
+    value: string | number;
+    breakdown: {
+        item_total: {
+            value: number | string;
+            currency_code: string;
+        };
+        shipping?: ItemTotal;
+    };
+}
+
+export interface AvaliableShippingOption {
+    additionalDescription: string;
+    cost: number;
+    id: string;
+    isRecommended: boolean;
+    description: string;
+}
+
+export interface PayerDetails {
+        payer: {
+            name: {
+                given_name: string;
+                surname: string;
+            };
+            email_address: string;
+            payer_id: string;
+            address: {
+                country_code: string;
+            };
+        };
+        purchase_units: PurchaseUnits[];
+}
+
+export interface ShippingChangeData {
+    amount: {
+        breakdown: {
+            item_total: ItemTotal;
+            shipping: ItemTotal;
+            tax_total: ItemTotal;
+        };
+        currency_code: string;
+        value: string;
+    };
+    orderID: string;
+    payment_token: string;
+    shipping_address: ShippingAddress;
+    selected_shipping_option: {
+        id: string;
+        amount: ItemTotal;
+    };
+}
+
+export interface ShippingAddress {
+    city: string;
+    postal_code: string;
+    country_code: string;
+    state: string;
+}
+
+export interface PaymentMethodInit {
+    initializationData: {
+        intent?: string;
+        isHosted?: boolean;
+    };
+}
+
+export interface CurrentShippingAddress {
+    city: string;
+    countryCode: string;
+    postalCode: string;
+}
+
+interface ItemTotal {
+    currency_code: string;
+    value: string;
+}
+
+export interface ShippingOptions {
+    id: string;
+    type: string;
+    label: string;
+    selected: boolean;
+    amount: ItemTotal;
+}
+
+export interface AddressData {
```

Comment:
This interface can be removed. You can add `email?: string` to `AddressRequestBody` interface instead of creating new interface

Reply:
No there are many required fields in AddressRequestBody, that i don't need to fill out in order to get response i need